libraries/common/components/Link/index.jsx

```diff
@@ -41,7 +41,12 @@ class Link extends Component {
     }
 
     const action = this.props.replace ? ACTION_REPLACE : ACTION_PUSH;
-    this.props.navigate(action, this.props.href, this.props.state);
+
+    this.props.navigate({
```

Comment:
You should provide two new functions here: `historyReplace` and `historyPush` and change the behavior.